storagetransfer/create_client_test.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2021 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+
+import backoff
+from google.api_core.exceptions import RetryError
+from googleapiclient.errors import HttpError
+import pytest
+
+import create_client
+import create_client_apiary
+
+
+@pytest.fixture()
+def job_filter(project_id: str):
+    yield json.dumps({'project_id': project_id})
+
+
+@backoff.on_exception(backoff.expo, (RetryError,), max_time=60)
+def test_create_client(job_filter: str):
+    client = create_client.create_transfer_client()
+
+    # a simple test to prove a usable client has been created.
+    # The output isn't relevant - just that a valid API call can be made
+    client.list_transfer_jobs({'filter': job_filter, 'page_size': 1})
```

Comment:
What happens if there is a usable client has not been created? Does it throw an exception: does it throw `RetryError` and we let `backoff` try and deal with it?

Reply:
Good question - I would expect an error to be raised to fail the test. I'll make note of this

Reply:
I would prefer that the error gets handled if we know its specifics, but fine to move on as-is for now since we already have the `@backoff.on_exception` decoration added.